### PR TITLE
ENT-9569: Apply the 60s SSL handshake timeout to the embedded Artemis server

### PR DESCRIPTION
**Pull request by shamsasari**
This ensures that any CRL connection timeouts due to unreachable dist points will be ignored if soft fail revocation is enabled. Otherwise, the default 10s SSL handshake timeout is too quick against the default 15s CRL timeout in the JDK. This was achieved by using a custom `AcceptorFactory` (`NodeNettyAcceptorFactory`) which sets the `handshakeTimeoutMillis` property on the Netty `SslHandler`.

A series of other change were made to facilate this change:
* `sslHandshakeTimeout` is now a `Duration` rather than long millis, which is helpful since the JDK CRL timeout is in seconds
* Any exceptions ignored due to soft fail are logged as debug
* The new acceptor factory also adds the Netty logging handler if `trace` is true
* The transport methods in `ArtemisTcpTransport` have been cleaned up and duplication removed. It’s now easier to understand which transports need to the SSL handshake timeout parameter
* `CertificationRevocationListNodeTests` has been cleaned up and duplication removed. It’s now easier to understand the various test cases. SSL handshake timeout tests were added for both the Artemis and the AMQP server classes
* lib/quasar.jar, which is meant to be git ignored, has been removed from git tracking

### CODE REVIEW

File: client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClient.kt [hunk 1]
```diff
@@ -61,8 +60,12 @@ class RPCClient<I : RPCOps>(
             sslConfiguration: ClientRpcSslOptions? = null,
             configuration: CordaRPCClientConfiguration = CordaRPCClientConfiguration.DEFAULT,
             serializationContext: SerializationContext = SerializationDefaults.RPC_CLIENT_CONTEXT
-    ) : this(rpcConnectorTcpTransport(haAddressPool.first(), sslConfiguration),
-            configuration, serializationContext, rpcConnectorTcpTransportsFromList(haAddressPool, sslConfiguration))
+    ) : this(
+            rpcConnectorTcpTransport(haAddressPool.first(), sslConfiguration),
+            configuration,
+            serializationContext,
+            haAddressPool.map { rpcConnectorTcpTransport(it, sslConfiguration, true) }
```
Comment (rick-r3) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClient.kt]: Why specify `true` here but not at line 64 ?

**Comment by shamsasari** [comment]
There's a bit of refactoring to make the code and test easier to understand. Let me know if you want to reduce or remove it. I will now check that this change is mergeable into ENT.

**Comment by shamsasari** [comment]
Otherwise this PR is now ready for review.

**Comment by shamsasari** [comment]
The test is failing due to what I believe is buggy soft fail exception reporting in the JDK. `java.security.cert.PKIXRevocationChecker#getSoftFailExceptions` returns ignored exceptions due to soft fail. I've noticed that it doesn't return anything on JDK 8 builds in the 200 build range. New ~300 builds work.

As advised by @adelel1 we need to keep to the old JDK 8 build, I will revert the extra checks in the test.

**Comment by shamsasari** [comment]
@rick-r3 `CertificateRevocationListNodeTests` had some refactoring done in ENT (https://github.com/corda/enterprise/pull/2135). I can make the equivalent changes here, they seem reasonable.

Otherwise the diff between 4.8 OS and ENT seems fine.

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**